pkg/agent/metrics/prometheus.go

```diff
@@ -104,19 +104,18 @@ func InitializePrometheusMetrics() {
 		klog.Errorf("Failed to retrieve agent K8S node name: %v", err)
 	}
 
-	gaugeHost := metrics.NewGauge(&metrics.GaugeOpts{
-		Name: "antrea_agent_runtime_info",
-		Help: "Antrea agent runtime info , defined as labels. The value of the gauge is always set to 1.",
-
+	deprecatedGaugeHost := metrics.NewGauge(&metrics.GaugeOpts{
```

Comment:
I would leave gauge names as they are - as renaming just adds plenty of unnecessary diffs and merge efforts later on.
Just add maybe DeprecatedVersion: "x.x" to the gauge defs, e.g
metrics.GaugeOpts{
	Name: "antrea_agent_runtime_info",
	Help:   "...",			
	ConstLabels:   "..."
	StabilityLevel: metrics.STABLE,
	DeprecationVersion: "x.x"}

Reply:
DeprecatedVersion is to implement the deprecation lifecycle for Kubernetes control plane metrics as given [here](https://github.com/kubernetes/enhancements/blob/master/keps/sig-instrumentation/20190404-kubernetes-control-plane-metrics-stability.md#deprecation-lifecycle). 
The instrumented version is Kubernetes version. We could employ a simpler policy as mentioned in https://github.com/vmware-tanzu/antrea/pull/1217#issuecomment-691282184  

I would like to not add this field to avoid the confusion of the deprecation policy and version numbers. 

Can you elaborate on merging difficulties with the change in the variable name? IMO, this serves as a quick indicator in source code that these metrics have to be removed in the future.

Reply:
> Hi @srikartati,
> See deprecation example [here](https://github.com/kubernetes/kubernetes/blob/release-1.17/pkg/proxy/metrics/metrics.go#L49).
> BTW that metric has been actually removed since then :)

In the above example given by you, there is a change in the variable name.

Reply:
Hi @srikartati, DeprecatedVersion should expose deprecation to users (AFAIK k8s metrics will note that in the metric help) while renaming variables is only visible to developers. I'm not sure that this is enough.
Not really sure which confusion you're trying to avoid by not using it.
Merging wise - that change will popup as a diff on every branch comparison...
My vote is still to remove these metrics immediately, but if we decide to deprecate in a later time, lets use the best tools for this purpose.